simpleviewinc/keg-herkin
# ZEN-557: Parkin Backend Integration

**Ticket**: [ZEN-557](https://jira.simpleviewtools.com/browse/ZEN-557)

## Context

* During development of ZEN-544, we discovered that `cucumber-jest` does not work well with cucumber-expressions (does not type cast 
* we could patch `cucumber-jest`, but since we already have our own implementation with the `Parkin` project, it's not really worth it to support two separate projects

## Goal

* Integrate parkin into keg-herkin
* Also update run task to support tag filters

## Updates

* `configs/aliases.config.js`
  * added alias to the tesUtils/parkin directory
* `configs/jest.parkin.config.js`
  * renamed from `jest.cucumber.config.js`
  * updated testMatcher to match with `scripts/runParkin.js`
* `container/mutagen.yml` && `container/values.yml`
  * added parkin mutagen sync
* `package.json`
  * added parkin dependency
* `repos/example/tests/bdd/steps/given.js`
  * updated example step definitions to use parkin's given statement
* `repos/testUtils/parkin/hooks.js`
  * exporting parkin hooks for access in step definitions
* `repos/testUtils/parkin/instance.js`
  * instantiating and exporting parkin instance
* `repos/testUtils/parkin/loadFeatures.js`
  * loads all feature files as specified by paths in herkin-config
* `repos/testUtils/parkin/runFeatures.js`
  * runs the feature files using parkin
  * warns the user if no features are found
* `repos/testUtils/parkin/stepFunctions.js`
  * exporting parkin step functions (Given, When, etc.) 
* `repos/testUtils/steps/**/*.js`
  * updating our pre-defined step definitions to use parkin
* `scripts/runParkin.js`
  * script that executes inside container when `keg herkin cucumber test` is run on host machine
  * runs all the feature tests
* `tasks/definitions/cucumber/run.js`
  * updated to work with parkin
* 

## Testing

* `keg herkin && keg pr 19`
* `keg herkin start --config ./repos/example/herkin.config.js --from keg-herkin:zen-557-parkin-integration`
* `keg herkin cucumber test --headless`
* verify tests pass
* `keg herkin cucumber test --name texas --headless`
* verify you see message indicating no features were run
* `keg herkin cucumber test --tags @texas --headless`
* verify you see message indicating no features were run
* `keg herkin cucumber test --tags @example --headless`
* verify tests pass
* `keg herkin cucumber test --name Example` 
* verify tests pass
